src/box/memtx_hash.cc

```diff
@@ -286,9 +286,20 @@ memtx_hash_index_random(struct index *base, uint32_t rnd, struct tuple **result)
 
 	*result = NULL;
 	uint32_t k = light_index_random(hash_table, rnd);
-	if (k == light_index_end)
+	struct txn *txn = in_txn();
+	struct space *space = space_by_id(base->def->space_id);
+	if (k == light_index_end) {
+		/*
+		 * `light_index_end` is returned only in case the space is
+		 * empty.
+		 */
+		memtx_tx_track_full_scan(txn, space, base);
 		return 0;
+	}
 	*result = light_index_get(hash_table, k);
+	assert(*result != NULL);
+	assert(!base->def->key_def->is_multikey);
```

Comment:
Does hash supports multikey?

Reply:
Please read this snippet:
https://github.com/tarantool/tarantool/blob/8ee0e434610189f3e4bbca0264de60aa79f0696e/src/box/memtx_space.c#L732-L737

Reply:
Yep, it doesn't. Do we need this assert then?

Reply:
Removed assertion (it's intent was actually to document the fact that **HASH** index cannot be multikey).